sdcflows/interfaces/bspline.py

```diff
@@ -288,116 +292,137 @@ def _run_interface(self, runtime):
-                    unwarp,  # if no HMC matrices, interpolator can be shared
-                    runtime.cwd,
-                )
-                for i, fname in enumerate(self.inputs.in_data)
-            ]
-        else:
-            # Embarrasingly parallel execution
-            from concurrent.futures import ProcessPoolExecutor
-
-            outputs = [None] * len(self.inputs.in_data)
-            with ProcessPoolExecutor(max_workers=self.inputs.num_threads) as ex:
-                outputs = ex.map(
-                    _b0_resampler,
-                    self.inputs.in_data,
-                    [self.inputs.in_coeff] * n,
-                    pe_dir,
-                    ro_time,
-                    hmc_mats,
-                    [None] * n,  # force a new interpolator for each process
-                    [runtime.cwd] * n,
-                )
+        data2fmap_xfm = None
+
+        if isdefined(self.inputs.data2fmap_xfm):
+            data2fmap_xfm = Affine.from_filename(
+                self.inputs.data2fmap_xfm if not isinstance(self.inputs.data2fmap_xfm, list)
+                else self.inputs.data2fmap_xfm[0],
+                fmt="itk"
+            ).matrix
+        elif isdefined(self.inputs.fmap2data_xfm):
+            # Same, but inverting direction
+            data2fmap_xfm = (~Affine.from_filename(
+                self.inputs.fmap2data_xfm if not isinstance(self.inputs.fmap2data_xfm, list)
+                else self.inputs.fmap2data_xfm[0],
+                fmt="itk"
+            )).matrix
+
+        # Pre-cached interpolator object
+        unwarp = B0FieldTransform(
+            coeffs=[nb.load(cname) for cname in self.inputs.in_coeff]
+        )
 
-        (
-            self._results["out_corrected"],
-            self._results["out_warp"],
-            self._results["out_field"],
-        ) = zip(*outputs)
+        # We can now write out the fieldmap
+        self._results["out_field"] = fname_presuffix(
+            self.inputs.in_data,
+            suffix="_field",
+            newpath=runtime.cwd,
+        )
 
-        out_fields = set(self._results["out_field"]) - set([None])
-        if len(out_fields) == 1:
-            self._results["out_field"] = out_fields.pop()
+        # HMC matrices are only necessary when reslicing the data (i.e., apply())
+        # Check the length of in_xfms matches that of in_data
+        self._results["out_corrected"] = fname_presuffix(
+            self.inputs.in_data,
+            suffix="_sdc",
+            newpath=runtime.cwd,
+        )
 
+        unwarp.apply(
+            self.inputs.in_data,
+            self.inputs.pe_dir,
+            self.inputs.ro_time,
+            xfms=self.inputs.in_xfms if isdefined(self.inputs.in_xfms) else None,
+            xfm_data2fmap=data2fmap_xfm,
+            approx=self.inputs.approx,
+            num_threads=(
+                self.inputs.num_threads if isdefined(self.inputs.num_threads) else None
+            ),
```

Comment:
less verbose
```suggestion
            xfms=self.inputs.in_xfms or None,
            xfm_data2fmap=data2fmap_xfm,
            approx=self.inputs.approx,
            num_threads=self.inputs.num_threads or None,
```

Reply:
nipype's `Undefined` doesn't evaluate to False, as far as I know.

Reply:
```
In [3]: from nipype.interfaces.base import Undefined

In [4]: bool(Undefined)
Out[4]: False
```